pkg/operation/botanist/controlplane.go

```diff
@@ -68,6 +68,8 @@ func (b *Botanist) DeployNamespace(ctx context.Context) error {
 			gardencorev1alpha1.BackupProvider: string(b.Seed.CloudProvider),
 			gardencorev1alpha1.SeedProvider:   string(b.Seed.CloudProvider),
 			gardencorev1alpha1.ShootProvider:  string(b.Shoot.CloudProvider),
+			// TODO: remove once the new shoot API is introduced, value should be fetched from spec.networking.type
```

Comment:
Why? I think it still makes sense to keep labels on the namespaces for the various provider types. Probably we should also add another label of the operating system config types, etc. It's useful for webhooks.

Reply:
I did not mean to remove the label from the namespace. I meant to pick up the value of the label from the shoot `spec.networking.type` section. Now it is Calico, but different shoots can choose different networking plugins right?

Reply:
Then please enhance the comment, I didn't get what was meant. Also, consider introducing `spec.networking` already today, this will ease the implementation at various places, I guess.